docs/1.getting-started/11.testing.md

```diff
@@ -5,36 +5,378 @@ navigation.icon: i-ph-check-circle-duotone
+      "<div id=\\"test-wrapper\\">
+        <div>This is an auto-imported component</div>
+        <div> I am a global component </div>
+        <div>Index page</div><a href=\\"/test\\"> Test link </a>
+      </div>"
+    `)
+})
+```
+
+#### `mockNuxtImport`
+
+`mockNuxtImport` allows you to mock Nuxt's auto import functionality. For example, to mock `useStorage`, you can do so like this:
+
+```ts
+import { mockNuxtImport } from '@nuxt/test-utils/runtime'
+
+mockNuxtImport('useStorage', () => {
+  return () => {
+    return { value: 'mocked storage' }
+  }
+})
+
+// your tests here
+```
+
+> **Note**: `mockNuxtImport` can only be used once per mocked import per test file. It is actually a macro that gets transformed to `vi.mock` and `vi.mock` is hoisted, as described [here](https://vitest.dev/api/vi.html#vi-mock).
 
-In each `describe` block where you are taking advantage of the `@nuxt/test-utils` helper methods, you will need to set up the test context before beginning.
+If you need to mock a Nuxt import and provide different implementations between tests, you can do it by creating and exposing your mocks using [`vi.hoisted`](https://vitest.dev/api/vi.html#vi-hoisted), and then use those mocks in `mockNuxtImport`. You then have access to the mocked imports, and can change the implementation between tests. Be careful to [restore mocks](https://vitest.dev/api/mock.html#mockrestore) before or after each test to undo mock state changes between runs.
+
+```ts
+import { vi } from 'vitest'
+import { mockNuxtImport } from '@nuxt/test-utils/runtime'
+
+const { useStorageMock } = vi.hoisted(() => {
+  return {
+    useStorageMock: vi.fn().mockImplementation(() => {
+      return { value: 'mocked storage'}
+    })
+  }
+})
+
+mockNuxtImport('useStorage', () => {
+  return useStorageMock
+})
+
+// Then, inside a test
+useStorageMock.mockImplementation(() => {
+  return { value: 'something else' }
+}) 
+```
+
+#### `mockComponent`
+
+`mockComponent` allows you to mock Nuxt's component.
+The first argument can be the component name in PascalCase, or the relative path of the component.
+The second argument is a factory function that returns the mocked component.
+
+For example, to mock `MyComponent`, you can:
+
+```ts
+import { mockComponent } from '@nuxt/test-utils/runtime'
+
+mockComponent('MyComponent', {
+  props: {
+    value: String
+  },
+  setup(props) {
+    // ...
+  }
+})
+
+// relative path or alias also works
+mockComponent('~/components/my-component.vue', async () => {
+  // or a factory function
+  return {
+    setup(props) {
+      // ...
+    }
+  }
+})
+
+// or you can use SFC for redirecting to a mock component
+mockComponent('MyComponent', () => import('./MockComponent.vue'))
+
+// your tests here
+```
+
+> **Note**: You can't reference to local variables in the factory function since they are hoisted. If you need to access Vue APIs or other variables, you need to import them in your factory function.
```

Comment:
```suggestion
> **Note**: You can't reference local variables in the factory function since they are hoisted. If you need to access Vue APIs or other variables, you need to import them in your factory function.
```